src/UI/App.jsx

```diff
@@ -13,23 +13,58 @@ import PropTypes from "prop-types";
 import Launcher from "./Launcher";
 import Chat from "./Chat";
 import Api from "../Api/Api";
-import {API_ACCOUNT_IDENTIFICATION, API_DEVICE_IDENTIFICATION, API_DOMAIN} from "./tempConfig";
 import ApiEventTarget from "../Api/ApiEventTarget";
 import {version} from "../../package.json";
 import PollingService from "../Api/Polling";
 import {messages} from "../Api/Constants/Events";
 import DeviceTypes from "../Api/Constants/DeviceTypes";
+import {ApiOptions, InterfaceTexts, InterfaceTextsContext} from "./Scripts/Context";
+import deepMerge from "deepmerge";
+import {Observable} from "object-observer";
+import deepForEach from "deep-for-each";
+import Logger from "js-logger";
+import {areWeOnline} from "./Scripts/WorkingHours";
+import {isiOSMobileDevice, isMobile} from "./Scripts/OSRecognition";
 
 export default class App extends React.Component {
 	constructor(props) {
 		super(props);
 
-		this.state = {showChat: false};
+		const interfaceLanguage = window?.parleySettings?.runOptions?.country || "en";
+		const interfaceTextsDefaults = interfaceLanguage === "nl" ? InterfaceTexts.dutch : InterfaceTexts.english;
+		this.state = {
+			showChat: false,
+			offline: false,
+			isMobile: isMobile(),
+			isiOSMobile: isiOSMobileDevice(),
+			interfaceLanguage,
+			interfaceTexts: {
+				...interfaceTextsDefaults,
+				title: window?.parleySettings?.runOptions?.interfaceTexts
+					?.desc || interfaceTextsDefaults.title,
+				inputPlaceholder: window?.parleySettings?.runOptions?.interfaceTexts
+					?.placeholderMessenger || interfaceTextsDefaults.inputPlaceholder,
+				sendingMessageFailedError: window?.parleySettings?.runOptions?.interfaceTexts
+					?.messageSendFailed || interfaceTextsDefaults.sendingMessageFailedError,
+				serviceUnreachableError: window?.parleySettings?.runOptions?.interfaceTexts
+					?.serviceUnreachableNotification || interfaceTextsDefaults.serviceUnreachableError,
+				welcomeMessage: window?.parleySettings?.runOptions?.interfaceTexts
+					?.infoText || interfaceTextsDefaults.welcomeMessage,
+			},
+			apiDomain: window?.parleySettings?.apiDomain || ApiOptions.apiDomain,
+			accountIdentification: window?.parleySettings?.roomNumber || ApiOptions.accountIdentification,
```

Comment:
Misschien een overbodige discussie, maar is het niet logischer dat de specifieker gezette optie (de ApiOptions context) de globale optie (de window.parleySettings) overruled ipv andersom?

Of kan dit niet omdat de defaults ook via die context binnekomen?

Reply:
Nee lijkt me niet handig.
De `ApiOptions` zijn defaults die we mee "shippen" in de code.
De `window.parleySettings` kunnen we niet mee geven aan de klant, dus die moeten ze zelf instellen.

Als `ApiOptions` altijd overruled dan kan je nooit meer `window.parleySettings` gebruiken tenzij je de `ApiOptions` uit de code gaat slopen.

Reply:
Ja het probleem is dat ApiOptions ook onze defaults bevat. Het liefste zou je deze hierarchie zien, waarbij ieder volgende de voorgaande overruled.

1. Default ApiOptions
2. window.parleySettings
3. Door implementeerder gezette ApiOptions

Ik dacht nu aan dat 3, 2 moet kunnen overrulen. Maar doordat je dan ook krijgt dat 1 nr 2 zal overrulen kan dit nu niet. Dan laten we het zo, want daar nog een oplossing voor vinden/maken kost tijd en is niet direct nodig.

Reply:
Prima, dan resolve ik deze conversatie